### ISSUE TITLE
Pre-allocate and reuse tensors

### ISSUE THREAD

**Issue by budgetdevv** [budgetdevv/FlorenceSharp#2]
Instead of letting the GC collect the managed tensors, pre-allocate them and reuse. Especially important for logits generation, where tensors are 50k+ floating point values. Unfortunately, the tensors are re-allocated each step
